packages/app/src/cli/models/extensions/specifications/marketing_activity_extension.ts

```diff
@@ -1,22 +1,28 @@
 import {MarketingActivityExtensionSchema} from './marketing_activity_extension_schemas/marketing_activity_extension_schema.js'
 import {createExtensionSpecification} from '../specification.js'
+import {randomUUID} from '@shopify/cli-kit/node/crypto'
 
 const spec = createExtensionSpecification({
-  identifier: 'marketing_activity_extension_cli',
+  identifier: 'marketing_activity',
   schema: MarketingActivityExtensionSchema,
   appModuleFeatures: (_) => ['bundling'],
   deployConfig: async (config, _) => {
     return {
       title: config.title,
       description: config.description,
-      app_api_url: config.app_api_url,
+      api_path: config.api_path,
       tactic: config.tactic,
       marketing_channel: config.marketing_channel,
       referring_domain: config.referring_domain,
       is_automation: config.is_automation,
       use_external_editor: config.use_external_editor,
       preview_data: config.preview_data,
-      fields: config.fields,
+      fields: config.fields.map((field) => ({
+        ...field,
+        // NOTE: we're not using this id anywhere, generating it to satisfy the schema
+        // decided not to remove it from the schema for now to minimize the risk of breaking changes
+        id: randomUUID(),
```

Comment:
This won't be indefinite. Given the tight timeline for the migration, we decided against changing the API here as it was referenced in a lot of places in the BE code so we wanted to avoid making that change for now. We do plan on deprecating this extension altogether in favour of a new extension, but there's no hard timeline for that yet. We're hoping that will happen sometime next year.